### ISSUE TITLE
runs in loop don't finish 

### ISSUE THREAD

**Issue by richarddwang** [wandb/client#1081]
* Weights and Biases version: 0.8.36
* Python version: 3.7
* Operating System: Ubuntu 16.04.6

The problem is if I run many runs in a loop, every runs will last to the end of program but not the beginning of its next run.

according to the advice of #951, I tried 
```
import time
for _ in range(2):
    run = wandb.init(reinit=True)
    with run:
        run.log({'metric':10})
    time.sleep(10)
```
But still I got this
![image](https://user-images.githubusercontent.com/17963619/83626267-c1663f80-a5c7-11ea-9ad1-091b27cfd225.png)

**Question**
- How can I make the run finished when init a new run ? (so it can record correct exec time)
- Does this mean if I run 15 run in a loop, I will get 15 threads exist at the same time ?

**Comment by tyomhak** [comment]
Hi, I believe you need to add wandb.join() at the end of the loop.
You can find more information about that in our docs, [here](https://docs.wandb.com/library/init#how-do-i-launch-multiple-runs-from-one-script).

**Comment by richardyy1188** [comment]
Hi, sorry for the late reply.
I tried the code in the doc:
```
import wandb
for x in range(10):
    wandb.init(project="runs-from-for-loop", reinit=True)
    for y in range (100):
        wandb.log({"metric": x+y})
    wandb.join()
```
and even
```
for x in range(10):
    run = wandb.init(project="runs-from-for-loop", reinit=True)
    with run:
        for y in range (100):
            wandb.log({"metric": x+y})
    wandb.join()
```

Both get me the same result
![image](https://user-images.githubusercontent.com/17963619/84033142-fceb7a00-a9ca-11ea-95c2-8313033febd0.png)
Only the last run will finish, and othe run even keep accumulate running time after the program stops.

**Comment by vanpelt** [comment]
@richardyy1188 I just want to confirm you're running this inside of a jupyter notebook?  We're aware of an issue with status reporting in Jupyter and are are working on a fix.

**Comment by richardyy1188** [comment]
Hi, @vanpelt , thanks for your reply.

Yes I did do it in a jupyter notebook, is there an issue I can follow to know when it is solved ?

**Comment by vanpelt** [comment]
We'll follow up here when it's fixed.

**Comment by onoderat** [comment]
Are there any updates on this issue? Does the "running" status create performance issues, and is there a temporary hack to induce say the "crash" status instead?

PS: Thanks for creating this wonderful software!

**Comment by raubitsj** [comment]
FYI: This is fixed in our pre-release (soon to be released) version:
`%pip install --upgrade --pre wandb`

**Comment by raubitsj** [comment]
The preferred syntax for the loop would be:
```
import wandb
for x in range(10):
    run = wandb.init(project="runs-from-for-loop", reinit=True)
    for y in range (100):
        run.log({"metric": x+y})
    run.join()
```

or:
```
for x in range(10):
    run = wandb.init(project="runs-from-for-loop", reinit=True)
    with run:
        for y in range (100):
            run.log({"metric": x+y})
```

**Comment by onoderat** [comment]
@raubitsj Thanks a tonne for the update - just confirmed that it is working.

**Comment by cwaubert** [comment]
After upgrading from 0.10.30 to 0.10.31 I am running into the same problem:
- end of run in not detected hence my program won't finish

**Comment by vanpelt** [comment]
@cwaubert can you try adding `wandb.finish()` in your loop or use a context block, i.e.

```
with wandb.init(...) as run:
   # do your stuff, call `run.log(...)` etc.
```

**Comment by cwaubert** [comment]
@vanpelt thanks for the quick response! That seems to work for single runs.
However, I am trying to run a sweep and here it won't change from one run to the next. Works like a charm with 0.10.30 though!

**Comment by vanpelt** [comment]
This sounds like a regression.  Can you send us a minimal example that reproduces this problem?

**Comment by bsridatta** [comment]
This seems to be resolved in 10.32
